File: Sample/Java/src/scov/TableSpawn.java
package scov;

import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import ScorpioProto.Commons.*;
import ScorpioProto.Table.*;
@SuppressWarnings("unused")

public class TableSpawn implements ITable<Integer, DataSpawn> {
	final String FILE_MD5_CODE = "484cdae7d179982f1c7868078204d81d";
    private int m_count = 0;
    private HashMap<Integer, DataSpawn> m_dataArray = new HashMap<Integer, DataSpawn>();
    public TableSpawn Initialize(String fileName, IScorpioReader reader) {
        int iRow = TableUtil.ReadHead(reader, fileName, FILE_MD5_CODE);
        for (int i = 0; i < iRow; ++i) {
            DataSpawn pData = DataSpawn.Read(fileName, reader);
            if (m_dataArray.containsKey(pData.ID())) {
                m_dataArray.get(pData.ID()).Set(pData);
            } else {
                m_dataArray.put(pData.ID(), pData);
            }
        }
        m_count = m_dataArray.size();
        return this;
    }
    
    public DataSpawn GetValue(Integer ID) {
        if (m_dataArray.containsKey(ID)) return m_dataArray.get(ID);
        TableUtil.Warning("DataSpawn key is not exist " + ID);
        return null;
    }

    public boolean Contains(Integer ID) {
        return m_dataArray.containsKey(ID);
    }

    public final HashMap<Integer, DataSpawn> Datas() {
        return m_dataArray;
    }
    
    public IData GetValueObject(Object ID) {
        return GetValue((Integer)ID);
    }

    public boolean ContainsObject(Object ID) {
        return Contains((Integer)ID);
    }

    public int Count() {
        return m_count;
    }
}

File: Sample/nodejs/Code/TableSpawn.js

const TableUtil = require("../ScorpioProto/TableUtil")
const DataSpawn = require("./DataSpawn").DataSpawn
class TableSpawn {
    constructor() {
        this.m_count = 0
        this.m_dataArray = {}
    }
    Initialize(fileName, reader) {
        let iRow = TableUtil.ReadHead(reader, fileName, "484cdae7d179982f1c7868078204d81d");
        for (let i = 0; i < iRow; ++i) {
            let pData = DataSpawn.Read(fileName, reader);
            if (this.Contains(pData.ID())) {
                this.m_dataArray[pData.ID()].Set(pData);
            } else {
                this.m_dataArray[pData.ID()] = pData;
            }
        }
        this.m_count = Object.getOwnPropertyNames(this.m_dataArray).length;
        return this;
    }
    GetValue(ID) {
        if (this.Contains(ID)) return this.m_dataArray[ID];
        //TableUtil.Warning("DataSpawn key is not exist " + ID);
        return null;
    }
    Contains(ID) {
        return this.m_dataArray[ID] != null;
    }
    Datas() {
        return this.m_dataArray;
    }
    GetValueObject(ID) {
        return this.GetValue(ID);
    }
    ContainsObject(ID) {
        return this.Contains(ID);
    }
    GetDatas() {
        return this.Datas();
    }
    Count() {
        return this.m_count;
    }
}
exports.TableSpawn = TableSpawn;
